### DISCUSSION TITLE
HA inaccessible to 99% of world, because automation editing is a nightmare. Need simple visual editor!

### DISCUSSION THREAD

**Discussion by MBarti** [home-assistant/frontend#8499] [Automations / scripts / scenes]
Current state:
99% of people (made up statistic) are dissuaded from using HA, because everyone says it's a steep learning curve, mostly due to a horrible automations editing experience. Everything elce can be set up without learning any coding/ linux commands.

A layman is faced with: entity.IDs, "calling services", filling in "service data" using proper formats, parentheses, numbers, etc.... It's not a language a normal user should know or be using.

So I see that almost all development effort is focused on fixing details for the 1% of population (the geeks) who already know HA. While noone is thinking of expanding HA to non-geeks, who just want a simple automations editor.

I'm sure this has been discussed before, but seriously, this is by far the biggest hurdle hampering HAs growth. Automations need a simple visual editor, with Icons, understandable language, big buttons .... you know, for normal people. And no, installing node-red, does not solve it.

What it should be:
No need to re-invent the wheel, can copy the UX flow of Apple Home app, or Eve Homekit app, etc., who've already spent years user-testing the UX.
https://www.macrumors.com/how-to/create-homekit-automations/

So the flow would be: 
1. Select trigger from a nice, visual, understandable list, big icons, simple language. [= don't show a full page full of forms to fill and weirdly-named fields, like it is now]
2. Select conditions (as clickable pre-filled fields / visual switches) - specific to the selected trigger
3. Select devices or scenes to be included in the automation (to visually click a checkmark next to them)
4. Set target state: Show all the devices in a visual way (clear icons + text), so the user can click, like in lovelace, to set the desired state (e.g. set lamp to 50% brightness by moving a slider [not filling out forms...]). 
DONE!
5. Write into config, so the user never ever has to know what a YAML is or how to properly close parentheses...
6. PROFIT and watch how HA gets adopted by thousands of new novice users worldwide!

**Comment by spacegaier** [comment]
Don't have time right now to dive into detail, although out of the gate I do disagree with a few mentioned points, e.g.. displaying a long list of all entities in an HA instance to scroll through and pick the ones you need sounds like a bad UX to me. I don't want to scroll through hundreds of entries to find the one I need. A picker as we have is much faster and easier.

The automation editor also hides most of the YAML already. You mostly need it for advanced stuff nowadays. If you have concrete proposals (or even better PRs) go ahead.

Are you aware of HA Blueprints? With those creating automation becomes even easier. Three simple inputs and you have motion activated lights for example.

![image](https://user-images.githubusercontent.com/114137/109644965-78431c00-7b56-11eb-8d0f-e17ba335a918.png)

**Comment by MBarti** [comment]
My point is, that I try to represent the silent majority who does not send comments on github:)

I think that HA is amazing and is incredibly close to becoming mass-market ready and accessible to non-technical people. Once everything is visual and one does not need to edit any files or fill any forms or learn any syntax, HA may become much more popular - getting the share of people that currently can handle a sophistication of tapping icons in an app.
There's no real reason, why HA couldn't be similar - it's all about working out the last kinks in the UX, especially in the automations area.

I saw this happen with the Homebridge project (a bridge to add non-certified accessories to Homekit) - once it got a visual UX plugin, it really took off wildly with lots of new people starting to use it, lots on pre-made devices with pre-installed software ant the UX plugin started to appear (Raspi based +pre-installed SD card + UX plugin), based on the promise of simplicity for non-techies.

Regarding your points - have you ever tried Homekit? Everyone can hate on Apple, but the fact is they spend countless hours simplifying thins for non-techies. Homekit is ridiculously easy to work with, yet provides very powerful tools in a non-overwhelming way. I have managed to set up a full home automation set-up without learning any syntaxes, edititing any configs, reading any documentation - it's natural, intuitive, 100% tap-friendly.

Regarding the display of long list of "entities" (btw. that's a technical word - again, requiring from the user the effort of learning of the lingo of HA...), you can display all _devices_ grouped by rooms - can easily fit >100 icons on 1 page (desktop), with simple scrolling for more. If there's a lot more, you can collapse the rooms. You can also add a "search" bar, which filers the icons once you start typing... if you're used to the current way.

I wish I coud do a PR, but I'm not a programmer. All I could help with is provide simple mock-ups of the flow... But the I'd just probably recreate the Homekit experience (or any other similarly easy one - like the Eve app on iOS).

Here's how to start:
1. Add new automation --> Show a screen like that:
<img width="485" alt="image" src="https://user-images.githubusercontent.com/22871912/110015883-af672800-7d24-11eb-92de-1cd1a68a9bc3.png">
(you can add additional option like "Initiated by another service" (webhook)", etc.)

2. ... if you can't access homekit, see a sample video with details of the flow: https://youtu.be/d9_6p1110lo?t=840 

I'm not trying to be condescending or an Apple fanboy, but it really pains me to see an awesome project like HA not get the mass market scale it deserves.

**Reply by frenck** [reply]
Those interfaces are possible, at the price of lesser options.

**Reply by MBarti** [reply]
Those interfaces are always possible. You can always add a small "Advanced options" button, for the expert users.
(e.g. Homekit adds a (i) icon, showing much more complex options, but hides them on purpose not to owerwhelm the 99% of people who don't need them). The UX as default should cater to the most people (i.e. non-techies), but with options for the pros.

**Reply by frenck** [reply]
> Those interfaces are always possible. You can always add a small "Advanced options" button, for the expert users.

We already have that, that would give you even more in Home Assistant 😂

**Reply by MBarti** [reply]
Why do I bother?:)

HA is the "linux" of home automation systems. Amazing possibilities, fully customizable, hardcore UX, dedicated know-it-all fanbase, ~1% market share ;)

**Comment by SeanPM5** [comment]
The HomeKit automations UX flow is indeed very nice and simple, but it’s also _extremely_ limited in scope compared to what Home Assistant allows. As frenck said, that simplicity is achieved by limiting the number of options.

In the HomeKit app you can't have triggers based on a sensors temperature, humidity, lux, battery level for example. There's only a handful of conditions that you can use. Actions are lacking - there's no support for delays, you can't even do basic things like make your HomePod speak text (tts). There's entire product categories like robot vacuums that aren't yet supported by HomeKit. And you can't even name your automations either.

There's some nice parts of the Home app UX for sure that could serve as inspiration. But it's not as straightforward as just "copying" them, because once you start adding in all the features they lack/hide, the interface won't remain that simple obviously.

**Comment by MBarti** [comment]
I strongly disagree. The simplicity of Homekit is achieved through human-centric design, not nerd-centric:)
Advanced options are there, but they only show up when they're relevant. 
Of course HA has more advanced options, but if well designed, they'd add 10% of complexity, to a UX that'd be 100x easier to use than the current one.

If HA requires a manual/ documentation to use it, it has failed in making the usability good enough. In the last 10 years I haven't seen any consumer-facing apps that require reading some documentation to use.

This design is utterly horrible for any UX designer:
![](https://www.home-assistant.io/images/getting-started/action.png)

Imagine explaining to your mother "Sooo, to turn on the light you have to CALL ACTIONS, then select the proper ENTITY IDs for the lights, but make sure the SERVICE selected is LIGHT(dot)TURN(underscore)ON. And make sure to fill in the form properly by issuing a command SERVICE_UNDERSCORE_ALL(colon):ALL, because you know, mom, the entities are the lights...)....; but if you wqnt different lights, then you need to find their entityIDs... simple"
What a piece of crap, seriously.

Anyway, taking a step back - I feel that 99% of development effort on HA is focused on satisfying the 1% of people who use it, while ignoring the silent majority who don't bother to speak up and try it.

If you have a specific example of an automation that could not conceivably and easily be fit into a Homekit-like automation flow, please share.

**Reply by spacegaier** [reply]
FYI: You are using an outdated HA version. Your screenshot no longer looks like this in the current version.

**Reply by frenck** [reply]
We'll at least agreed on the screenshot provided 👍  As it has been replaced.

**Reply by MBarti** [reply]
Great first step replacing the screenshot. Next step: rethinking the whole flow from a non-techy-user perspective :)

**Reply by frenck** [reply]
> Great first step replacing the screenshot

If you have followed the development a bit more closely, you would know this was not a first step at all, and rather offending for all the contributors that helped out where we are today.

**Reply by MBarti** [reply]
Well, I appreciate the quality work, but on the other hand some poor design choices deserve to be offended.

**Reply by frenck** [reply]
> poor design choices deserve to be offended

I don't think that is called for. Something does not get from point a -> b at once. Considering where it is coming from, I think a lot of people have done a great job to get to something more user-friendly compared to some years ago.

But hey, it is an open-source project. Jump in and make it better, because that is how it works.

**Comment by ludeeus** [comment]
73.6% of all statistics are made up

**Comment by MBarti** [comment]
Anyway, my points still stand.
I challenge anyone to provide an example of a common automation, that couldn't be fit into this clean, usable and non-technical Homekit-like automation flow.
